cmd/el-proxy/main.go

```diff
@@ -0,0 +1,482 @@
+			// As long as no further sequence numbers were assigned we can just decrement this.
+			p.nextSequenceNumber--
+		} else {
+			// TODO We need to implement gap filling here to handle parallel failed txs.
+			log.Warn("Request failed but sequence number not able to be rolled back")
+		}
+		p.seqNumAssignmentMutex.Unlock()
+		return common.Hash{}, fmt.Errorf("Error forwarding msg: %w", err)
+	}
+
+	return goWrapper.Transaction.Hash(), nil
+}
+
+//// We need to proxy some other methods for tools like cast to use when building txs.
+
+func (p *ExpressLaneProxy) ChainId(_ context.Context) hexutil.Uint64 {
+	chainId := p.config.ChainId
+	return (hexutil.Uint64)(chainId)
+}
+
+func (p *ExpressLaneProxy) GetTransactionCount(ctx context.Context, address common.Address, blockNumOrHash rpc.BlockNumberOrHash) (hexutil.Uint64, error) {
+	client, err := GetClientFromURL(ctx, p.config.RPCURL, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	var result hexutil.Uint64
+	err = client.CallContext(ctx, &result, "eth_getTransactionCount", address, blockNumOrHash)
+	return result, err
+}
+
+func (p *ExpressLaneProxy) FeeHistory(ctx context.Context, blockCount hexutil.Uint64, lastBlock rpc.BlockNumber, rewardPercentiles []float64) (json.RawMessage, error) {
+	var result json.RawMessage
+
+	client, err := GetClientFromURL(ctx, p.config.RPCURL, nil)
+	if err != nil {
+		return result, err
+	}
+
+	err = client.CallContext(ctx, &result, "eth_feeHistory", blockCount, lastBlock, rewardPercentiles)
+	return result, err
+}
+
+func (p *ExpressLaneProxy) BlockNumber(ctx context.Context) (uint64, error) {
+	client, err := GetClientFromURL(ctx, p.config.RPCURL, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	return ethclient.NewClient(client).BlockNumber(ctx)
+}
+
+func (p *ExpressLaneProxy) GetBlockByNumber(ctx context.Context, blockNum *rpc.BlockNumber, includeTxData bool) (json.RawMessage, error) {
+	var result json.RawMessage
+
+	client, err := GetClientFromURL(ctx, p.config.RPCURL, nil)
+	if err != nil {
+		return result, err
+	}
+
+	err = client.CallContext(ctx, &result, "eth_getBlockByNumber", blockNum, includeTxData)
+	return result, err
+}
+
+func (p *ExpressLaneProxy) GetTransactionReceipt(ctx context.Context, txHash hexutil.Bytes, opts *json.RawMessage) (json.RawMessage, error) {
+	log.Debug("Received eth_getTransactionReceipt", "txHash", txHash)
+
+	var result json.RawMessage
+	client, err := GetClientFromURL(ctx, p.config.RPCURL, nil)
+	if err != nil {
+		return result, err
+	}
+
+	err = client.CallContext(ctx, &result, "eth_getTransactionReceipt", txHash, opts)
+	return result, err
+
+}
+
+func mainImpl() int {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	args := os.Args[1:]
+	expressLaneProxyConfig, err := parseExpressLaneProxyArgs(ctx, args)
+	if err != nil {
+		confighelpers.PrintErrorAndExit(err, printSampleUsage)
+		panic(err)
+	}
+	stackConf := DefaultExpressLaneProxyStackConfig
+	stackConf.DataDir = "" // ephemeral
```

Comment:
I'm not familiar enough with how DataDir is used. But, does it need to be cleaned up on exit, or is it okay just to leave the directory around?